daemon/logger/splunk/splunk.go

```diff
@@ -228,12 +229,23 @@ func New(info logger.Info) (logger.Logger, error) {
 		}
 	}
 
+	var timeout time.Duration
+	if timeoutStr, ok := info.Config[splunkHTTPTimeout]; ok {
+		value, err := strconv.ParseInt(timeoutStr, 10, 64)
+
+		if err != nil {
+			return nil, err
+		}
+		timeout = time.Duration(value) * time.Millisecond
```

Comment:
Thanks for updating. Erm, I thought I commented, but looks like I never did 😬 

Similar to my comment on https://github.com/moby/moby/pull/43100#discussion_r774508995, I'm wondering if we should make the `splunk-timeout` option accept a "duration" string, instead of millisecond. Let me quote that comment here, as it may clarify why I think that would be good to do;

> Perhaps we should consider to parse the value as a `time.Duration` ([`time.ParseDuration()`](https://pkg.go.dev/time#ParseDuration)) to make it easier (and less error-prone) for the user to set the right interval. Having to specify the value in `ms` works "ok(ish)" if you want to, say, specify a second (`1000 ms`), but can become a bit awkward after that. It's also error-prone, for example, if a user wants to specify `1 second`, but specifies `1` and ends up with `1 ms`.
> 
> And looking at the code, I see we already use duration strings for `fluentd-retry-wait`, so it's a bit more consistent.
> 
> So I would suggest to;
> 
> * use `time.ParseDuration()`
> * validate the no negative values are specified
> * perhaps also a minimum / maximum (not sure what reasonable values are? between 100ms and 10s ? (open to suggestions)
> 
> Would be good to have a minimal unit test (I see there's currently no tests in this package, which is a bit unfortunate, but let's not continue that "tradition" 😅)


Happy to hear your thoughts!

Reply:
Of course, that sounds like a good suggestion. Sorry for missing it. I will get to it asap.

Reply:
I have added the duration string support and am testing for negative values. I didn't enforce min/max, given that the default is no timeout, it doesn't make sense to bound max value.